PyInstaller/__main__.py

```diff
@@ -146,7 +147,7 @@ def generate_parser() -> _PyiArgumentParser:
     return parser
 
 
-def run(pyi_args=None, pyi_config=None):
+def run(pyi_args: list | None = None, pyi_config: dict | list | None = None):
```

Comment:
Technically it can be an `Iterable[tuple[str, object]]` (as it's used to update a dict). But as you mentioned in another comment, doesn't mean that's what it's intended for. So I've updated it and other annotations meant to update `CONF` as well.